data/tribes/scripting/help/immovable_help.lua

```diff
@@ -30,15 +30,16 @@ include "tribes/scripting/help/format_help.lua"
 function immovable_help_string(tribe, immovable_description)
    local helptexts = immovable_description:helptexts(tribe.name)
    local result = ""
-   local purpose = helptexts["purpose"]
    local image = immovable_description.icon_name
-   if purpose ~= nil then
+   if helptexts.purpose then
```

Comment:
Actually this is what we call an implicit bool conversion in our c++ code.
I am happy to accept these in lua, however if they are bad in c++ we might think to fix them in lua as well.